corgi/collectors/brew.py

```
@@ -760,11 +762,123 @@ def get_builds_with_tag(
+                build_id=build_id,
+                defaults={"nvr": srpm},
+            )
+            # Lookup the rpm build_ids
+            rpms = [self.sans_epoch(rpm) for rpm in rhel_modules[srpm]]
+            rpm_lookup_calls = self.brew_rpm_lookup(rpms)
+            for rpm, call in rpm_lookup_calls:
+                srpm_build_id = call.result["build_id"]
+                srpm, _ = CollectorSRPM.objects.get_or_create(build_id=srpm_build_id)
+                rpm_obj, _ = CollectorRPM.objects.get_or_create(nvra=rpm, srpm=srpm)
+                rpm_obj.rhel_module.add(rhel_module)
+
+            yield build_id
+
+    def lookup_build_ids(self, rpm_filenames_by_srpm) -> Generator[str, None, None]:
+        # For each srpm look up it's build id
+        srpms = [srpm.removesuffix(".src") for srpm in rpm_filenames_by_srpm.keys()]
+        find_build_id_calls = self.brew_srpm_lookup(srpms)
+        for srpm, call in find_build_id_calls:
+            build_id = call.result
+            if not build_id:
+                for filename in rpm_filenames_by_srpm[srpm]:
+                    logger.debug(
+                        "Didn't find build with NVR %s, using rpm filename: %s",
+                        srpm,
+                        filename,
+                    )
+                    # We don't use a multicall here, because this won't be called
+                    # in most cases
+                    rpm_data = self.koji_session.getRPM(filename)
+                    if not rpm_data:
+                        # Try the next srpm rpm filename
+                        continue
+                    build_id = rpm_data["build_id"]
+                    # found the build_id, stop iterating filenames
+                    break
+                # if no filenames had RPM data
+                if not build_id:
+                    logger.warning("Unable to find build_id for %s when saving %s", srpm, build_id)
```

@jasinner This is probably the case you mentioned about generators returning None instead of a build_id. Maybe we should raise an exception here, or if not we can ```continue``` to the next srpm, call in find_build_id_calls.

---

yeah, let's continue.